controllers/mysqlcluster_controller.go

```diff
@@ -1862,6 +1858,51 @@ func (r *MySQLClusterReconciler) finalizeV1(ctx context.Context, cluster *mocov1
 	return nil
 }
 
+func (r *MySQLClusterReconciler) UpdateStatus(ctx context.Context, req ctrl.Request, cluster *mocov1beta2.MySQLCluster) error {
+	log := crlog.FromContext(ctx)
+
+	var sts appsv1.StatefulSet
+	err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.PrefixedName()}, &sts)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get StatefulSet %s/%s: %w", cluster.Namespace, cluster.PrefixedName(), err)
+	}
+
+	needUpdate := false
+	updated := metav1.ConditionFalse
+	reason := "UpdateInProgress"
+	if sts.Status.AvailableReplicas != 0 && sts.Status.CurrentRevision == sts.Status.UpdateRevision && sts.Generation == sts.Status.ObservedGeneration {
```

Comment:
MOCO may use partition in the future, so it is better to consider partition.

https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/polymorphichelpers/rollout_status.go#L138

Reply:
`UpdateRevision` will not be changed even when partitioned pods would have been updated. It would be changed when all of the pods in StatefulSet would be updated.
So comparing `CurrentRevision` and `UpdateRevision` does not affect even if partition is used.
But message like "waiting for partitioned roll out" is useful. So I will add this message in `reason`.